### PR TITLE
Remove unnecessary scheduling features

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
As of now even though following are beta upstream, I don't see much value in enabling the following features.

- `ResourceQuotaScopeSelectors` will be GA in 1.15. In the current state, it's not quite useful for us.
- `TaintNodesByCondition` has a 5%(approximate number) chance that node controller and kubelet will race. There are folks who started using this in upstream but this is something that needs some thorough discussion.
- `TaintBasedEvictions` is something that we wanted to use internally for draining but we haven't started using it. So, wanted to postpone it to next release.

/cc @sjenning @derekwaynecarr

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
> were any of these features on in 3.11?

None, taintNodesByCondition graduated to beta 1.12 and taintBasedEviction in 1.13

File: config/v1/types_feature.go [hunk 1]
```diff
@@ -74,6 +74,9 @@ var FeatureSets = map[FeatureSet]*FeatureGateEnabledDisabled{
 		},
 		Disabled: []string{
 			"LocalStorageCapacityIsolation", // sig-pod, sjenning
+			"ResourceQuotaScopeSelectors",   // sig-pod, ravig
+			"TaintNodesByCondition",         // sig-pod, ravig
```
[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply config/v1/types_feature.go]: Checking...

[the remark this answers is not in the document]
Comment (gnufied) [reply config/v1/types_feature.go]: Yeah shutdown node taint does not depend on these features yet.

**Comment by sjenning** [comment]
i'm on-board with this, but want final approval from @derekwaynecarr

**Comment by sjenning** [comment]
actually lets go ahead with this
/lgtm
/approve

**Comment by deads2k** [comment]
> actually lets go ahead with this
> /lgtm
> /approve

This pull broke e2e tests when brought into kube-apiserver and kube-scheduler

**Comment by ravisantoshgudimetla** [comment]
@deads2k - Perhaps it's because of ScheduleDS pods is enabled too, we want to have discussion on Monday if we want to ship TaintNodesByCondition or not. I created this PR which got merged an hour ago - https://github.com/openshift/api/pull/290

**Pull request merged**